### ISSUE TITLE
Solution of "Extend BookRestController" not conform to OpenAPI Spec

### ISSUE THREAD

**Issue by rweisleder** [workshops-de/Bookshelf_sample_solution#74]
With task "Extend BookRestController", the attendees are supposed to add a new mapping, so that the following mappings are present in the `BookRestController` afterwards:
```java
@GetMapping
public List<Book> getAllBooks() {
    // ...
}

@GetMapping("/{isbn}")
public Book getSingleBook(@PathVariable String isbn) throws Exception {
    // ...
}

@GetMapping(params = "author")
public Book searchBookByAuthor(@RequestParam String author) throws Exception {
    // ...
}
```

This means that for GET requests with path "/book" there are 2 mappings available now. For Spring this is not a problem. But if we add Springdoc/Swagger later, it gives a weird result:
![image](https://github.com/workshops-de/Bookshelf_sample_solution/assets/1654369/747438a0-20c6-4b3a-9b3a-7eb8c5ea86e1)

For "GET /book" we have only one operation where the query parameter `author` is required and which returns only one `Book` object. The operation `getAllBooks` returning multiple books does not appear anymore. (Attendees also ask every time why `searchBookByAuthor` returns only one book and not multiple.)

From my point of view, our `BookRestController` is no longer OpenAPI conform, because there only one operation may be defined per path and method, see [Specification](https://spec.openapis.org/oas/v3.1.0#path-item-object). I suggest to move `searchBookByAuthor` to another path.

**Comment by BjoernKW** [comment]
Yes, this is correct. `BookRestController` doesn't comply withe the OpenAPI specification. However, adding another path probably would make the code more complex.

Could you maybe provide a pull request implementing the structure you have in mind?

**Comment by bkratz** [comment]
Ich antworte einfach mal in Deutsch  😄 
Die "billigste" Lösung wäre wahrscheinlich, @RequestParam(required=false) zu verwenden. Dann darf man aber auch keine Validatoren wie @NotNull oder @NotBlank verwenden.
